File: Longest Substring Without Repeating Characters.js
var lengthOfLongestSubstring = function(s) {
    var seen = new Map();
    var start = 0, max = 0;
    for(let i=0; i<s.length; i++){
        if(seen.has(s[i])) start = Math.max(seen.get(s[i])+1, start);
        seen.set(s[i], i);
        max = Math.max(max, i-start+1)
    }
    return max;
};

File: Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        unique_chars = set()
        size = len(s)
        max_len = 0
        left = 0

        for right in range(size):
            while s[right] in unique_chars:
                unique_chars.remove(s[left])
                left += 1

            unique_chars.add(s[right])
            max_len = max(max_len, right - left + 1)

        return max_len
